### ISSUE TITLE
Receive OTLP HTTP metrics, forward to Prometheus, collision on metric names

### ISSUE THREAD

**Issue by marcalff** [open-telemetry/opentelemetry-collector-contrib#21471]
Setup:
* C++ application (test app, opentelemetry-cpp 1.9.0, OTLP HTTP exporter)
* opentelemetry-collector 0.75.0
* prometheus 2.42.0

In a test application, define meters and metrics as follows:

* Meter `my_meter_1`
  * Metric `my_metric_a`
  * Metric `my_metric_b`
* Meter `my_meter_2`
  * Metric `my_metric_a`
  * Metric `my_metric_b`
  * Metric `my_metric_c`

As far as I understand, this is valid, because what identifies a metric is the instrumentation scope + the metric name, so that `my_meter_1.my_metric_a` and `my_meter_2.my_metric_a` are two different metrics.

Export this using OTLP HTTP, to an opentelemetry-collector.

From the collector logs, the data received is:

```
2023-04-17T20:40:58.805+0200	info	MetricsExporter	{"kind": "exporter", "data_type": "metrics", "name": "logging", "#metrics": 5}
2023-04-17T20:40:58.805+0200	info	ResourceMetrics #0
Resource SchemaURL: 
Resource attributes:
     -> telemetry.sdk.version: Str(1.9.0)
     -> telemetry.sdk.language: Str(cpp)
     -> service.version: Str(xxx)
     -> telemetry.sdk.name: Str(opentelemetry)
     -> service.instance.id: Str(xxx)
     -> service.name: Str(xxx)
     -> service.namespace: Str(xxx)
ScopeMetrics #0
ScopeMetrics SchemaURL: 
InstrumentationScope my_meter_1 1.0
Metric #0
Descriptor:
     -> Name: my_metric_b
     -> Description: description b
     -> Unit: 
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
StartTimestamp: 2023-04-17 18:40:37.784020265 +0000 UTC
Timestamp: 2023-04-17 18:40:58.697529476 +0000 UTC
Value: 110
Metric open-telemetry/opentelemetry-collector#1
Descriptor:
     -> Name: my_metric_a
     -> Description: description a
     -> Unit: 
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
StartTimestamp: 2023-04-17 18:40:37.784020265 +0000 UTC
Timestamp: 2023-04-17 18:40:58.697529476 +0000 UTC
Value: 108
ScopeMetrics open-telemetry/opentelemetry-collector#1
ScopeMetrics SchemaURL: 
InstrumentationScope my_meter_2 1.0
Metric #0
Descriptor:
     -> Name: my_metric_b
     -> Description: description b
     -> Unit: 
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
StartTimestamp: 2023-04-17 18:40:37.784020265 +0000 UTC
Timestamp: 2023-04-17 18:40:58.697764025 +0000 UTC
Value: 116
Metric open-telemetry/opentelemetry-collector#1
Descriptor:
     -> Name: my_metric_a
     -> Description: description a
     -> Unit: 
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
StartTimestamp: 2023-04-17 18:40:37.784020265 +0000 UTC
Timestamp: 2023-04-17 18:40:58.697764025 +0000 UTC
Value: 114
Metric open-telemetry/opentelemetry-collector#2
Descriptor:
     -> Name: my_metric_c
     -> Description: description c
     -> Unit: 
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
StartTimestamp: 2023-04-17 18:40:37.784020265 +0000 UTC
Timestamp: 2023-04-17 18:40:58.697764025 +0000 UTC
Value: 118
	{"kind": "exporter", "data_type": "metrics", "name": "logging"}
```

So far, this works as expected: the opentelemetry collector receives data that describe:
* a measurement for instrumentation scope `my_meter_1` and metric `my_metric_a`
* another measurement for instrumentation scope `my_meter_2` and metric `my_metric_a`

Now, in the opentelemetry collector, forward the metric data to prometheus:

```
receivers:
  otlp:
    protocols:
      http:

processors:
  batch:
  memory_limiter:
    # 75% of maximum memory up to 4G
    limit_mib: 1536
    # 25% of limit up to 2G
    spike_limit_mib: 512
    check_interval: 5s

exporters:
  logging:
    loglevel: debug

  # Forward metrics to Prometheus
  prometheusremotewrite:
    endpoint: "http://localhost:9090/api/v1/write"
    # For official Prometheus (e.g. running via Docker)
    # endpoint: 'http://prometheus:9090/api/v1/write'
    # tls:
    #   insecure: true

service:
  pipelines:
    metrics:
      receivers: [otlp]
      processors: [memory_limiter, batch]
      exporters: [logging, prometheusremotewrite]
```

Data is sent using prometheusremotewrite, and is visible in prometheus.

The issue is that prometheus only displays:
* my_metric_a
* my_metric_b
* my_metric_c

The fully qualified name (my_meter_1.my_metric_a) is lost in the chain, so that data from different meters (my_meter_1 and my_meter_2) is mixed when there is a name collision on the metric name (my_meter_a).

Expected result:
* `my_meter_1_my_metric_a` visible in prometheus (or something similar)
* `my_meter_2_my_metric_a` visible in prometheus 

In other words, there should be some way to preserve the fully qualified name of a metric when exporting it from OTLP to prometheus, so that metrics from different meters can be separated.

**Comment by marcalff** [comment]
Familiar neither with the code nor with go, so I can't provide a patch.

From code reading, it seems the naming translation happens at:

* File `pkg/translator/prometheus/normalize_name.go`
* Function `BuildPromCompliantName()`
* Function `normalizeName()`

**Comment by marcalff** [comment]
@Aneurysm9 @rapphil @kovrus 

Ping to avoid automatically closing the issue, the problem is still relevant.

If this is tracked by another issue number, please link it.

**Comment by marcalff** [comment]
Again:

@Aneurysm9 @rapphil @kovrus

Ping to avoid automatically closing the issue, the problem is still relevant.

If this is tracked by another issue number, please link it.

In light of:

* https://github.com/open-telemetry/opentelemetry-specification/pull/3684
* https://github.com/open-telemetry/opentelemetry-specification/pull/3648

I would like to know how an application is expected to instrument metrics here:

* Should the metric name be globally unique, avoiding the collision ?
* Is it ok to have the same metric name provided by different meters, and if so how is the prometheus exporter handling the collision ?
